### PR TITLE
Add and automate a load lib test case for module zos_copy

### LINKED ISSUE #601: [Bug] The copy of a load module from PDSE to a PDSE failed
### Bug description

When we try to copy a load module from PDSE to a PDSE zos_copy failed with this message.

```
fatal: [zos_host_01]: FAILED! => {"changed": false, "failed_when_result": true, "msg": "Unable to copy data set member NAZARE.WDEPLOY.CORE140.LOADLIB(LGIPOL01) to data set member NAZARE.WDEPLOY.CORE140.BACK.LOADLIB(LGIPOL01)", "rc": 1, "stderr": null, "stderr_lines": null, "stdout": "cp: FSUM8976 Error writing \"//'NAZARE.WDEPLOY.CORE140.LOADLIB(LGIPOL01)'\" to PDSE member \"//'NAZARE.WDEPLOY.CORE140.BACK.LOADLIB(LGIPOL01)'\".: EDC5000I No error occurred.\ncp: !//'NAZARE.WDEPLOY.CORE140.BACK.LOADLIB(LGIPOL01)': close failed: EDC5091I The requested function could not be performed because a system utility failed.\n", "stdout_lines": null}
```

### Playbook verbosity output

TASK [ibm.ibm_zos_wazi_deploy.zos_deploy : [deploy_modules>update>member_archive]: Copy NAZARE.WDEPLOY.CORE140.LOADLIB(LGIPOL01) to NAZARE.WDEPLOY.CORE140.BACK.LOADLIB(LGIPOL01)] ***
fatal: [zos_host_01]: FAILED! => {"changed": false, "failed_when_result": true, "msg": "Unable to copy data set member NAZARE.WDEPLOY.CORE140.LOADLIB(LGIPOL01) to data set member NAZARE.WDEPLOY.CORE140.BACK.LOADLIB(LGIPOL01)", "rc": 1, "stderr": null, "stderr_lines": null, "stdout": "cp: FSUM8976 Error writing \"//'NAZARE.WDEPLOY.CORE140.LOADLIB(LGIPOL01)'\" to PDSE member \"//'NAZARE.WDEPLOY.CORE140.BACK.LOADLIB(LGIPOL01)'\".: EDC5000I No error occurred.\ncp: !//'NAZARE.WDEPLOY.CORE140.BACK.LOADLIB(LGIPOL01)': close failed: EDC5091I The requested function could not be performed because a system utility failed.\n", "stdout_lines": null}

### Contents of `ansible.cfg`

```YAML
[defaults]
forks = 25
host_key_checking = False
#callbacks_enabled = timer, profile_tasks, profile_roles
stdout_callback = ibm.ibm_zos_wazi_deploy.cb_evidences

[ssh_connection]
pipelining = True
#ssh_args = -o ControlMaster=auto -o ControlPersist=3600s -o PreferredAuthentications=publickey
```

### Contents of the inventory

_No response_

### Contents of `group_vars` or `host_vars`

_No response_

### Ansible version

```shell
ansible [core 2.11.12]
  config file = None
  configured module search path = ['/home/jby/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  ansible collection location = /home/jby/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.10 (default, Jun 22 2022, 20:18:18) [GCC 9.4.0]
  jinja version = 3.1.2
  libyaml = True
```

### IBM z/OS Ansible core Version

v1.4.0

### IBM ZOAU version

v1.2.1

### z/OS version

2.4

### Ansible module

zos_copy

### PR DESCRIPTION
**Pull request by fernandofloresg**
##### SUMMARY

Fixed error when a load lib member was copied to another load lib. Apparently error is found in stdout instead of err.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

zos_copy

Fixes #601 

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by fernandofloresg** [comment]
Screenshot 2023-03-22 at 1 44 05 PM

**Comment by fernandofloresg** [comment]
I'm working on creating a test case for our test suite https://github.com/ansible-collections/ibm_zos_core/issues/682 , but opened the PR to not delay the release if we decide to release this

**Comment by fernandofloresg** [comment]
Screenshot 2023-03-27 at 1 18 32 PM
Screenshot 2023-03-27 at 1 19 20 PM

File: changelogs/fragments/601-copy-loadlib-member.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+- zos_copy - Copy failed from a loadlib member to another loadlib member. Fix now looks for error in stdout in the if statement to use -X option.
```
Comment (rexemin) [inline changelogs/fragments/601-copy-loadlib-member.yml:2]: I think the fragment could be reworded to something like:
Fixes a bug in a conditional statement that caused a failure when copying a loadlib member to another one.

**Review by rexemin — APPROVED** [review]
Looks good, thanks for going through all the trouble to add the test.

File: changelogs/fragments/601-copy-loadlib-member.yml (hunk 1 above)
Comment (fernandofloresg) [reply changelogs/fragments/601-copy-loadlib-member.yml:2]: I had a similar wording originally, but went this route to follow ansible development guidelines for fragments: (https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) "Use past tense to describe the problem and present tense to describe the fix."

Comment (ddimatos) [reply changelogs/fragments/601-copy-loadlib-member.yml:2]: I did not even know they had a recommended guideline for this, learned something new.

File: tests/functional/modules/test_zos_copy_func.py [hunk 2]
```diff
@@ -1683,6 +1771,93 @@ def test_copy_pds_member_with_system_symbol(ansible_zos_module,):
...
+    cobol_pds = "USER.COBOL.SRC"
+    try:
+        hosts.all.zos_data_set(
+            name=src,
+            state="present",
+            type="pdse",
+            record_format="U",
+            record_length=0,
+            block_size=32760,
+            space_primary=40,
```
Comment (ddimatos) [inline tests/functional/modules/test_zos_copy_func.py:1790]: I know this is a result of the playbook I provided and no idea why i had the space_primary so large, I don't think any of these will really consume 40 MB, although not a show stopper right now , eventually we might want to be a bit more conservative with space. This is in all three data sets but ill comment here only.

File: tests/functional/modules/test_zos_copy_func.py [hunk 3]
```diff
@@ -1683,6 +1771,93 @@ def test_copy_pds_member_with_system_symbol(ansible_zos_module,):
...
+        assert rc == 0
+        rc = link_loadlib_from_cobol(hosts, src_name, cobol_pds)
+        assert rc == 0
+
+        copy_res = hosts.all.zos_copy(
+            src="{0}({1})".format(src, member), 
+            dest="{0}({1})".format(dest, "MEM1"), 
+            remote_src=True)
+
+        verify_copy = hosts.all.shell(
```
Comment (ddimatos) [inline tests/functional/modules/test_zos_copy_func.py:1838]: While this is a good check, one thing I think this test case is missing is did the load module copy successfully as in it still executable after its copied. This is easy to test, I believe I also provided JCL in the sample that is i github. I don't consider this a show stopper but would be nice if we knew it actually copied well enough to retain its exactable bits. 

Here are two additional ways to JCL to run the program:
```
  tasks:
    - name: Execute with mvscmd
      ansible.builtin.shell: mvscmd --pgm=HELLOPGM  --steplib=ANSIBLE.COBOL.LOADLIB --sysprint=* --stderr=* --stdout=*
      register: result

    - name: Result
      debug:
         msg: "{{result}}"

    - name: Execute program
      zos_mvs_raw:
        program_name: HELLOPGM
        auth: false
        verbose: true
        dds:
          - dd_data_set:
              dd_name: steplib
              data_set_name: ANSIBLE.COBOL.LOADLIB
              disposition: shr
          - dd_output:
              dd_name: sysprint
              return_content:
                type: text
          - dd_output:
              dd_name: stderr
              return_content:
                type: text
          - dd_output:
              dd_name: stdout
              return_content:
                type: text
      register: result

    - name: Result
      debug:
         msg: "{{result}}"
```

**Review by ddimatos — APPROVED** [review]
@fernandofloresg thanks for doing this, its great to see this test get covered. I did make a couple of comments, one about data set size of 40MB and the validation of the copied exactable by actually running it. I don't consider either a reason not to merge but we should at least create an issue to tune this a bit if we don't address it now.

File: tests/functional/modules/test_zos_copy_func.py (hunk 2 above)
Comment (fernandofloresg) [reply tests/functional/modules/test_zos_copy_func.py]: Thanks for the recommendation. Reduced it to 2MB per dataset.

File: tests/functional/modules/test_zos_copy_func.py (hunk 3 above)
Comment (fernandofloresg) [reply tests/functional/modules/test_zos_copy_func.py:1851]: Thanks, added the check. Only checks that rc == 0, if it cannot execute it rc != 0.

**Comment by fernandofloresg** [comment]
Screenshot 2023-04-12 at 4 07 29 PM

Screenshot 2023-04-12 at 4 07 44 PM

**Review by rexemin — APPROVED** [review]
Looks good, learned something new about changelog fragments.

**Review by ddimatos — APPROVED** [review]
Thanks for the updates @fernandofloresg , looks great, we were in serious need of this automated test.

**Comment by ddimatos** [comment]
I restored issue #601, and updated the fields for 1.5.0, 1.6.0 and put it in iteration 1 to be done , thank you.

**Pull request merged**